### PR TITLE
[core] Introduce product IT case

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
### Purpose

Introduce product IT case

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/PreAggregationITCase.java [hunk 1]
```diff
@@ -853,6 +853,157 @@ public void testStreamingRead() {
         }
     }
 
+    /** ITCase for product aggregate function. */
+    public static class ProductAggregation extends CatalogITCaseBase {
+        @Override
+        protected List<String> ddl() {
+            return Collections.singletonList(
+                    "CREATE TABLE IF NOT EXISTS T1 ("
+                            + "j INT, k INT, "
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/PreAggregationITCase.java:862]: Why not just set a single primary key?

Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/PreAggregationITCase.java:862]: 图片
Refer to the implementation of sum

Comment (yuzelin) [reply paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/PreAggregationITCase.java:862]: Alright.

**Review by yuzelin — APPROVED** [review]
+1

**Pull request merged**